### ISSUE TITLE
Returning a date object throws?

### ISSUE THREAD

**Issue by dblock** [amzn/ion-js#334]
This was a surprise to me:

```
        //The Ion specification allows year only precision, Javascript's Date does not, Ion has infinite precision decimal seconds whereas Javascript's Date rounds to milliseconds.
        //We have opted to allow users to generate Date objects within try catch statements so they can handle the specification differences as they see fit.
        //This is a one way lossy translation for user utility, do not use a Date to roundtrip data or expect any support for lossiness within the Date class and its APIs.
        if(this.precision === Precision.YEAR || this.precision === Precision.SECONDS) {
            let error =  new Error(this.precision === Precision.YEAR ? "Year" : "Seconds");
            error['date'] = date;
            throw error;
        }
```

First, relying on try/catch to control flow is a definite anti-pattern.

Secondly, if I want to shoot myself in the foot, I'd like to be able to without a try/catch, currently I am doing `Date.parse(reader.value().stringValue());`, which seems worse.

**Comment by pbcornell** [comment]
Agreed, this is an anti-pattern.

**Comment by pbcornell** [comment]
Resolved by #361 -- `Timestamp.getDate()` no longer throws a return value.

Also note that `Timestamp.stringValue()` is now `Timestamp.toString()`.
